pyannote/audio/tasks/joint_task/speaker_diarization_and_embedding.py

```diff
@@ -0,0 +1,1032 @@
+    >>> protocol = registry.get_protocol(...)
+
+    instantiate task
+    >>> task = JointSpeakerDiarizationAndEmbedding(protocol)
+
+    instantiate multi-task model
+    >>> model = JointSpeakerDiarizationAndEmbeddingModel()
+    >>> model.task = task
+
+    train as usual...
+
+    """
+
+    def __init__(
+            self,
+            protocol,
+            duration: float = 5.0,
+            max_speakers_per_chunk: int = 3,
+            max_speakers_per_frame: int = 2,
+            batch_size: int = 32,
+            database_ratio : float = 0.5,
+            num_workers: int = None,
+            pin_memory: bool = False,
+            augmentation: BaseWaveformTransform = None
+    ) -> None:
+        super().__init__(
+            protocol,
+            duration=duration,
+            batch_size=batch_size,
+            num_workers=num_workers,
+            pin_memory=pin_memory,
+            augmentation=augmentation,
+        )
+
+        self.max_speakers_per_chunk = max_speakers_per_chunk
+        self.max_speakers_per_frame = max_speakers_per_frame
+        self.database_ratio = database_ratio
+
+
+        # keep track of the use of database available in the meta protocol
+        # * embedding databases are those with global speaker label scope
+        # * diarization databases are those with file or database speaker label scope
+        self.embedding_database_files = []
+        self.diarization_database_files = []
+
+    def get_file(self, file_id):
+
+        file = dict()
+
+        file["audio"] = str(self.audios[file_id], encoding="utf-8")
+
+        _audio_info = self.audio_infos[file_id]
+        _encoding = self.audio_encodings[file_id]
+
+        sample_rate = _audio_info["sample_rate"]
+        num_frames = _audio_info["num_frames"]
+        num_channels = _audio_info["num_channels"]
+        bits_per_sample = _audio_info["bits_per_sample"]
+        encoding = str(_encoding, encoding="utf-8")
+        file["torchaudio.info"] = AudioMetaData(
+            sample_rate=sample_rate,
+            num_frames=num_frames,
+            num_channels=num_channels,
+            bits_per_sample=bits_per_sample,
+            encoding=encoding,
+        )
+
+        return file
+
+    def setup(self, stage="fit"):
+        """Setup method
+
+        Parameters
+        ----------
+        stage : {'fit', 'validate', 'test'}, optional
+            Setup stage. Defaults to 'fit'.
+        """
+
+        # duration of training chunks
+        # TODO: handle variable duration case
+        duration = getattr(self, "duration", 0.0)
+
+        # list of possible values for each metadata key
+        metadata_unique_values = defaultdict(list)
+
+        metadata_unique_values["subset"] = Subsets
+
+        if isinstance(self.protocol, SpeakerDiarizationProtocol):
+            metadata_unique_values["scope"] = Scopes
+
+        elif isinstance(self.protocol, SegmentationProtocol):
+            classes = getattr(self, "classes", list())
```

Comment:
Can probably be removed. Double check, though.

Reply:
> Can probably be removed. Double check, though.

Done in 3d295dd